fearless_simd_core/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+// Copyright 2025 the Fearless_SIMD Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! An abstraction to allow safely running custom `#[target_feature]` functions on stable Rust.
+//!
+//! This crate introduces the [`trampoline!`] macro, which allows running code in a
+//! statically validated `#[target_feature(enable="some_features")]` environment, based on
+//! externally provided tokens.
+//! This abstraction is designed to be combined with target features 1.1, the recent update
+//! in the Rust compiler to allow calling `#[target_feature]` functions safely from within
+//! other `#[target_feature]` functions.
+//! As such, once you have used the [`trampoline!`] macro, you can call any intrinsic in [`core::arch`].
+//!
+//! This crate also has modules which contain tokens for each Rust target features.
+//! These allow safely validating that a target feature is available, and obtaining a token.
+//! These are grouped by architecture:
+//!
+//! - [`x86`] contains the tokens for both the x86 and x86-64 targets.
+//!   It also contains tokens for each x86-64 microarchitecture level, see [`x86::V1`] for details.
+//! <!-- TODO: Other architectures -->
+//!
+//! # Examples
+//!
+//! At the time of writing, it is not possible to turn scalar values into <abbr title="Single InstructionMultiple Data">SIMD</abbr>
+//! vector types safely using only the standard library.
+//! These examples use [bytemuck](https://crates.io/crates/bytemuck) for this.
+//!
+//! <!-- TODO -->
```

Comment:
Is this TODO deliberately left for completing in a later PR?

Reply:
Yeah, that was my approximate intention. I'll plan to follow up with that very soon after we land this, but I don't want to potentially block this entire PR on getting those reviewed now.